compositor_api/src/types/from_register_input.rs

```diff
@@ -106,10 +110,20 @@ impl TryFrom<RtpInput> for pipeline::RegisterInputOptions {
                 .as_ref()
                 .map(|video| {
                     Ok(input::rtp::InputVideoStream {
-                        options: match video {
-                            InputRtpVideoOptions::FfmepgH264 => decoder::VideoDecoderOptions {
+                        options: match video.decoder {
+                            VideoDecoder::FfmpegH264 => decoder::VideoDecoderOptions {
                                 decoder: pipeline::VideoDecoder::FFmpegH264,
                             },
+
+                            #[cfg(feature = "vk-video")]
+                            VideoDecoder::VulkanVideo => decoder::VideoDecoderOptions {
+                                decoder: pipeline::VideoDecoder::VulkanVideoH264,
+                            },
+
+                            #[cfg(not(feature = "vk-video"))]
+                            VideoDecoder::VulkanVideo => {
+                                return Err(TypeError::new(NO_VULKAN_VIDEO))
+                            }
```

Comment:
We may want to ship compiled binaries with this feature enabled (at least for Linux x86). Scripts for generating binaries should be updated to handle that (this feature is not enabled by default if I understand it correctly).

What happens when someone tries to enable this feature on a device that doesn't support the vulkan video? Does it crash on compile, on input register or in run-time? If compilation has to be done on a device with vulkan video support, we might not be able to build those binaries using GH Actions.

Reply:
We should ship only one version. For macos it would be disabled, and for linux always be enabled. But yes good point release script probably needs to be updated, I don't see anywhere where this feature was defined so I'm not sure if it is enabled by default

Reply:
not sure if feature is needed, maybe just change it based on OS

Reply:
If we want to remove the feature we can use something like this to define an alias for the OS support `cfg` line: https://lib.rs/crates/cfg_aliases

I would like to keep the feature in though, maybe some users want to experiment with weird configurations if moltenvk adds support for vulkan video